### PR TITLE
call buyTicket when registration finishes or buy button is clicked + beautify code

### PR DESCRIPTION
**Pull request by tomerlevi444**

### CODE REVIEW

File: Example/KlarnaOnDemandExample/MainViewController.m [hunk 1]
```diff
@@ -18,17 +18,71 @@ - (void)viewDidLoad {
...
+  if([self hasUserToken]) {
+    [self buyTicket];
+  }
+  else {
+    [self openKlarnaRegistration];
+  }
+}
+
+- (IBAction)onChangePaymentPressed:(id)sender {
+  [self openKlarnaPreferences];
```
Comment (yuval-netanel) [inline Example/KlarnaOnDemandExample/MainViewController.m]: This method call other method, so I think that there is no need for the other method

Comment (grozen) [inline Example/KlarnaOnDemandExample/MainViewController.m]: :+1:

File: Example/KlarnaOnDemandExample/MainViewController.m [hunk 2]
```diff
@@ -18,17 +18,71 @@ - (void)viewDidLoad {
...
+  // send order request to app-server.
+  [self performPurchaseOfItemWithReference:@"TCKT0001" originProof:originProof];
+}
+
 - (void)performPurchaseOfItemWithReference:(NSString *)reference originProof:(KODOriginProof *)originProof {
   AFHTTPRequestOperationManager *manager = [AFHTTPRequestOperationManager manager];
   manager.requestSerializer = [AFJSONRequestSerializer serializer];
   NSString *userToken = [self getUserToken];
-
+
```
Comment (grozen) [inline Example/KlarnaOnDemandExample/MainViewController.m]: Mon dieu!

File: Example/KlarnaOnDemandExample/MainViewController.m [hunk 3]
```diff
@@ -18,17 +18,71 @@ - (void)viewDidLoad {
...
   NSString *userToken = [self getUserToken];
-
+  
   NSDictionary *params = @{
                            @"origin_proof" : [originProof description],
                            @"reference" :    reference,
                            @"user_token" :   userToken
                            };
-
+
```
Comment (grozen) [inline Example/KlarnaOnDemandExample/MainViewController.m]: Sacrebleu!

**Comment by yuval-netanel** [comment]
Approved!

**Comment by grozen** [comment]
Approved!

File: Example/KlarnaOnDemandExample/MainViewController.m (hunk 1 above)
Comment (tomerlevi444) [inline Example/KlarnaOnDemandExample/MainViewController.m]: Done.

File: Example/KlarnaOnDemandExample/MainViewController.m (hunk 2 above)
Comment (tomerlevi444) [inline Example/KlarnaOnDemandExample/MainViewController.m]: Fixed.

File: Example/KlarnaOnDemandExample/MainViewController.m (hunk 3 above)
Comment (tomerlevi444) [inline Example/KlarnaOnDemandExample/MainViewController.m]: Fixed.

**Pull request merged**